File: taobao-shopcart-interface/src/main/java/com/taobao/shopcart/service/ShopCartService.java
package com.taobao.shopcart.service;

import com.taobao.pojo.group.Cart;
import com.taobao.pojo.group.CartItem;

import java.util.List;

public interface ShopCartService {


    /**
     * 更新购物车
     *
     */
    void updateCart(Cart cart);

    /**
     * 得到用户购物车
     */
    Cart getCart(Long userId);

    /**
     * 删除redis中提交的购物项的列表
     */

    void delete(List<Long> idList, Long userId);
}

File: taobao-user-web/src/main/webapp/js/service/shopCartService.js
app.service('shopCartService', function ($http) {

    this.addToCart = function (cartItem) {
        return $http.post("shopCart/addToCart.do",cartItem);
    }

    //得到购物车的最后一个购物项则为刚才添加的商品项
    this.getLast = function () {
        return $http.get("shopCart/getLast.do");
    }

    this.findAll = function () {
        return $http.get("shopCart/findAll.do");
    }
    //批量删除
    this.del = function (ids) {
        return $http.get("shopCart/delete.do?ids="+ids);

    }
});